### PR TITLE
Recover from changing preference Imperial

### PR DESCRIPTION
**Pull request by davidakachaos**
When switching from metric to imperial and back, Validator now resets the report, removes highlights and recreates its tab.

### CODE REVIEW

File: src/basic.js [hunk 1]
```diff
@@ -727,3 +727,14 @@ function onNodesRemoved(e) {
...
+
+/**
+* Recover from switching isImperial pref
+*/
+function onChangeIsImperial() {
+	clearReport();
+	_RT.$HLedObjects = {};
+	_RT.$HLlayer.destroyFeatures();
+	_RT.$isMapChanged = true;
+	async(F_INIT);
```
[the remark this answers is not in the document]
Comment (davidakachaos) [reply src/basic.js]: We need to restore the tab. I could take the code that initializes the tab and UI, make it its own function and run only that. Would that be better?

**Comment by davidakachaos** [comment]
@berestovskyy I now use F_LOGIN. Needed to add a safeguard to prevent double tabs, but this seems to work correctly.

File: src/basic.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (berestovskyy) [reply src/basic.js]: Do we need to destroy features and objects here then? I mean we do reset all the dictionary during initialization, aren't we?

[the remark this answers is not in the document]
Comment (davidakachaos) [reply src/basic.js]: That is true. Do we do the same in F_LOGIN? Will check on that later 😃

**Comment by berestovskyy** [comment]
I got a report in PM that switching from imperial to metric is not convenient especially on the borders between US (imperial) & Mexico (metric). So I guess we should come up with a solution based on the address of a segment, not the settings. What do you think?

**Comment by davidakachaos** [comment]
Hmmm, that would be difficult, but I'll think on it.

Op do 25 okt. 2018 om 19:33 schreef Andriy Berestovskyy <
notifications@github.com>:

**Comment by berestovskyy** [comment]
I thought we could leave the speed checks as is by default, but (for quite a few countries) we could override the regexp to check speed limits in miles without any conversion, i.e. put there those strange km/h numbers as allowed.

**Comment by davidakachaos** [comment]
I'm looking at the `getLocalizedValue` function for options. I know we had in a previous iteration a exception check on certain countries, but we then choose to make it so it would depend on the `W.model.isImperial` setting.
Maybe the check on what country the segment is in wasn't that bad (ugly but it worked?)

**Comment by davidakachaos** [comment]
Btw, this PR only resolves the issue when someone is switching between the "Metrical" and "Imperial" setting. Currently the Validator tab disappears, this PR prevents that from happening, so I think I should open a new PR for this fix.

And yes! I think passing the country to `getLocalizedValue` and checking on that if we need the mph or the kph is a solid way to go. That would solve the speed validation problem. Do you happen to have a PL on the border between the US and Mexico for example to test this?

**Comment by berestovskyy** [comment]
Ok, so we need this change anyway, so let's merge it.

I was told a border part of Mexico (metric country) is both in USA and ROW servers. Here are the links:

ROW: https://www.waze.com/en-US/editor/?zoom=5&lat=32.63744&lon=-115.49456&env=row&segments=511258484

USA: https://www.waze.com/en-US/editor/?zoom=5&lat=32.63744&lon=-115.49456&env=usa&segments=511258484

**Comment by davidakachaos** [comment]
Opened a new PR for the fix, so merging this.

**Pull request merged**